src/dmd/cond.d

```diff
@@ -149,15 +151,25 @@ extern (C++) final class StaticForeach : RootObject
         sc = sc.endCTFE();
         el = el.optimize(WANTvalue);
         el = el.ctfeInterpret();
+        Expressions *es;
         if (el.op == TOK.int64)
         {
-            dinteger_t length = el.toInteger();
-            auto es = new Expressions();
-            foreach (i; 0 .. length)
+            if (aggr.isArrayLiteralExp())
             {
-                auto index = new IntegerExp(loc, i, Type.tsize_t);
-                auto value = new IndexExp(aggr.loc, aggr, index);
-                es.push(value);
+                // Directly use the elements of the array for the TupleExp creation
+                auto ale = cast(ArrayLiteralExp)aggr;
```

Comment:
No need for a cast here.
```
if (auto ale = aggr.isArrayLiteralExp())
```

Reply:
Of course, I suppose my first `if` had more than one condition.